namer/consul/src/main/scala/io/buoyant/namer/consul/SvcAddr.scala

```diff
@@ -85,6 +87,9 @@ private[consul] object SvcAddr {
                 " Last known state is %s",
               datacenter, key.name, e, currentValueToLog
             )
+            if (e.toString.contains(NonExistentDcMessage)) {
```

Comment:
I think we probably want to include all instances of UnexpectedResponse here.  That should cover more cases and be less brittle.

Reply:
Just for my curiosity, Is there a case where we wouldn't want to update the state when a HTTP 500 error occurs? If we received a 500 other than an invalid datacenter, should we update the state? I briefly looked through the consul agent source code and one error that might return a 500 is `errRPCRateExceeded`. Which I assume is rate limiting. I would think we would want to retry and not update the state to Addr.Neg since a subsequent request would generally end up being a HTTP 200. I may be totally wrong though.

Regardless, I understand that updating the Addr state to `Neg` is not permanent and eventually it would be updated to `Ok` once we get a good value from consul.

Reply:
Good question.  Unfortunately, consul doesn't give us a good machine readable way to differentiate.  Either way, you are correct that the retry loop would continue and the state would be updated.